Server/src/services/vehicle.service.ts

```diff
@@ -98,7 +102,12 @@ export default class VehicleService {
 			// now it is unlikely, that weights can be added to the app logs, but we could at least try it
 			logger.warn(`Could not add weights to tracker logs for vehicle with id ${vehicle.uid}.`)
 		} else {
-			const tempWeightedTrackKm = await this.weightedLogsToWeightedTrackKm(weightedTrackerLogs, track)
+			const tempWeightedTrackKm = await this.weightedLogsToWeightedTrackKm(
```

Comment:
The if-case above seems to be impossible, as addWeightToLogs always returns a list and never null